src/libraries/Microsoft.PowerFx.Core/Glue/SimpleResolver.cs

```diff
@@ -48,27 +51,44 @@ internal class SimpleResolver : INameResolver
 
         public IExternalEntity CurrentEntity => null;
 
-        public SimpleResolver(IEnumerable<EnumSymbol> enumSymbols, params TexlFunction[] extraFunctions)
-            : this(extraFunctions)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SimpleResolver"/> class.
+        /// </summary>
+        /// <param name="config"></param>
+        public SimpleResolver(PowerFxConfig config)            
         {
-            _enums = enumSymbols.ToArray();
-        }
+            _config = config ?? throw new ArgumentNullException(nameof(config));
 
-        public SimpleResolver(params TexlFunction[] extraFunctions)
-        {
-            _library = GetFullFunctionLibrary(extraFunctions);
-        }
-
-        public static TexlFunction[] GetFullFunctionLibrary(TexlFunction[] extraFunctions)
-        {
-            var list = new List<TexlFunction>();
-            list.AddRange(BuiltinFunctionsCore.BuiltinFunctionsLibrary);
-            list.AddRange(extraFunctions);
-            return list.ToArray();
+            _library = config.Functions.ToArray();
+            _enums = config.EnumStore.WithRequiredEnums(_library).EnumSymbols.ToArray();
```

Comment:
Actually we could make a new function here that returns the list of enums from the EnumStore as an array, augmented with the required enums, without actually modifying the EnumStore instance.

_enums = config.EnumStore.ToArrayWithRequiredEnums(_library);